kaniko/executor_kubernetes.go

```diff
@@ -176,44 +176,58 @@ func (e KubernetesExecutor) Execute(ctx context.Context, args []string) error {
 			},
 		},
 	}
-	err = mergeObjectWithYaml(&podTemplate, e.JobConfig.PodTemplateOverride)
+	err = mergeObjectWithYaml(&pod, e.PodConfig.PodTemplateOverride)
 	if err != nil {
 		return err
 	}
 
-	job := batchv1.Job{
-		TypeMeta: metav1.TypeMeta{
-			APIVersion: "batch/v1",
-			Kind:       "Job",
-		},
-		ObjectMeta: objectMeta,
-		Spec: batchv1.JobSpec{
-			BackoffLimit: pointer.Int32Ptr(0),
-			Template:     podTemplate,
-		},
+	_, err = e.clientSet.CoreV1().Pods(e.PodConfig.Namespace).Create(ctx, &pod, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to create kaniko pod: %w", err)
 	}
 
-	_, err = e.clientSet.BatchV1().Jobs(e.JobConfig.Namespace).Create(ctx, &job, metav1.CreateOptions{})
-	if err != nil {
-		return fmt.Errorf("failed to create kaniko job: %w", err)
+	if err := printPodLog(ctx, e.clientSet, e.PodConfig.Namespace, name); err != nil {
+		return err
 	}
 
-	return watchJob(ctx, e.clientSet, e.JobConfig.Namespace, name)
+	// Getting the pod status might require a few retries
+	for i := 0; i < 5; i++ {
+		podResult, err := e.clientSet.CoreV1().Pods(e.PodConfig.Namespace).Get(ctx, pod.Name, metav1.GetOptions{})
```

Comment:
Is it possible to use the Watch API ? I'm afraid this could often fail if the pod takes time to be created (for instance if the cluster needs to upscale, or if the image takes a very long time to pull.
```go
clientset.CoreV1().Pods(namespace).Watch(metav1.ListOptions{
    LabelSelector: labels,
})
```
I don't know if it will be testable, so an other lead is to use the Events API to watch for specific events for the pod